x-pack/plugins/security/public/management/roles/edit_role/privileges/kibana/privilege_form_calculator/privilege_form_calculator.ts

```diff
@@ -95,18 +94,25 @@ export class PrivilegeFormCalculator {
    * @param featureId the feature id
    * @param privilegeIndex the index of the kibana privileges role component
    */
-  public getEffectivePrimaryFeaturePrivilege(featureId: string, privilegeIndex: number) {
+  public getEffectivePrimaryFeaturePrivilege(
+    featureId: string,
+    privilegeIndex: number,
+    allSpacesSelected?: boolean
+  ) {
     const feature = this.kibanaPrivileges.getSecuredFeature(featureId);
 
     const basePrivilege = this.getBasePrivilege(privilegeIndex);
 
     const selectedFeaturePrivileges = this.getSelectedFeaturePrivileges(featureId, privilegeIndex);
 
-    return feature
+    const displayedPrivilege = feature
       .getPrimaryFeaturePrivileges({ includeMinimalFeaturePrivileges: true })
       .find((fp) => {
         return selectedFeaturePrivileges.includes(fp.id) || basePrivilege?.grantsPrivilege(fp);
       });
+    const correctSpacesSelected = displayedPrivilege?.requireAllSpaces ? allSpacesSelected : true;
+    const availablePrivileges = correctSpacesSelected && !displayedPrivilege?.disabled;
+    if (availablePrivileges) return displayedPrivilege;
```

Comment:
The phrase "displayed privilege" means something different, so we should probably rename this to avoid confusion:

```suggestion
    const effectivePrivilege = feature
      .getPrimaryFeaturePrivileges({ includeMinimalFeaturePrivileges: true })
      .find((fp) => {
        return selectedFeaturePrivileges.includes(fp.id) || basePrivilege?.grantsPrivilege(fp);
      });
    const correctSpacesSelected = effectivePrivilege?.requireAllSpaces ? allSpacesSelected : true;
    const availablePrivileges = correctSpacesSelected && !effectivePrivilege?.disabled;
    if (availablePrivileges) return effectivePrivilege;
```